pipelines/imputation/testing/CompareVcfs.wdl

```diff
@@ -5,35 +5,45 @@ workflow CompareVcfs {
     input {
         File test_gvcf
         File truth_gvcf
+
+        Int memory_gb = 64
     }
 
     call CompareVcfs {
         input:
             file1 = test_gvcf,
             file2 = truth_gvcf,
-            patternForLinesToExcludeFromComparison = "^##"
+            patternForLinesToExcludeFromComparison = "##" # ignore headers but do compare sample names
     }
 }
 
 task CompareVcfs {
     input {
         File file1
         File file2
-        String patternForLinesToExcludeFromComparison
+
+        String patternForLinesToExcludeFromComparison = ""
+
+        Int memory_gb = 64
     }
+
     Int disk_size_gb = ceil(10 * size(file1, "GiB")) + ceil(10 * size(file2, "GiB")) + 50
+
     command {
         set -eo pipefail
 
-        gunzip -c -f ~{file1} | grep -v '~{patternForLinesToExcludeFromComparison}' > file_1.vcf
-        gunzip -c -f ~{file2} | grep -v '~{patternForLinesToExcludeFromComparison}' > file_2.vcf
-
-        diff -q file_1.vcf file_2.vcf
+        if [ -z "~{patternForLinesToExcludeFromComparison}" ]; then
+            diff <(gunzip -c -f ~{file1}) <(gunzip -c -f ~{file2})
+        else
+            echo "patternForLinesToExcludeFromComparison is defined: '~{patternForLinesToExcludeFromComparison}'"
+            diff <(gunzip -c -f ~{file1} | grep -v '~{patternForLinesToExcludeFromComparison}') <(gunzip -c -f ~{file2} | grep -v '~{patternForLinesToExcludeFromComparison}')
+        fi
     }
 
     runtime {
-        docker: "gcr.io/gcp-runtimes/ubuntu_16_0_4:latest"
+        docker: "gcr.io/gcp-runtimes/ubuntu_16_0_4@sha256:025124e2f1cf4d29149958f17270596bffe13fc6acca6252977c572dd5ba01bf"
         disks: "local-disk ${disk_size_gb} SSD"
-        memory: "64 GiB"
+        memory: "${memory_gb} GiB"
+        preemptible: 3
```

Comment:
i would say we prob dont want preemptibles here.  this is something we prob just want to run asap would be my guess unless it runs super fast (which i dont expect)

Reply:
sure, though remember this is effectively a copy of the actual test warp does, so this one will only be run manually in which case i'd think we care less about ASAP and wouldn't mind the $ savings

Reply:
i can make it an input to the wdl so that the user can decide

Reply:
yea if its comparing large vcfs most likely being preemptible wouldnt be a cost saving.  being parameterized sounds good

Reply:
sounds good and i'll set the default to 0 (here and in the other one)